bftengine/include/bftengine/Replica.hpp

```diff
@@ -48,6 +49,7 @@ class IRequestsHandler {
                       concordUtils::SpanWrapper &parent_span) = 0;
 
   virtual void onFinishExecutingReadWriteRequests() {}
+  virtual void setControlStateManager(std::shared_ptr<IControlStateManager> controlStateManager) {}
```

Comment:
If the class is abstract, why don't we make the method abstract too?
The same applies to `IReplica`.

Reply:
There are some derived classes that don't need to implement this method, I prefer to give a default empty implementation than forcing every derived class to implement this method

Reply:
If some derived classes do not need the method then the method is not supposed to be declared in the base class.
Adding a method to the interface class is a very strong statement.
Can it be added only to specific implementations?

Reply:
You right, but the problem is that all upper-level references/pointers are for the base class.
I can make this method abstract of course and add an empty implementation where needed, but I'm afraid it may break API with concord and things like that...
Do you think we should do it in any case?

Reply:
As far as I see the method `setControlStateManager` is used only by the classes derived from `ICommandsHandler`.
I think it would be right to add move the method there and declare it abstract.
Regarding the product, I think the only class to be changed is `ConcordCommandsHandler`.
WDYT?